src/seedu/addressbook/commands/FavouriteCommand.java

```diff
@@ -0,0 +1,43 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.common.Messages;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList.PersonNotFoundException;
+import seedu.addressbook.data.person.UniquePersonList.AlreadyFavouritedException;
+
+/**
+ * Favourites a person identified using it's last displayed index from the address book.
+ */
+public class FavouriteCommand extends Command {
+
+    public static final String COMMAND_WORD = "favourite";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Favourites the person identified by the index number used in the last person listing.\n"
+            + "Parameters: INDEX\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_FAVOURITE_PERSON_SUCCESS = "Favourited %1$s";
+    public static final String MESSAGE_PERSON_ALREADY_FAVOURITED = "Peron is already favourited";
+
+    public FavouriteCommand(int targetVisibleIndex) {
+        super(targetVisibleIndex);
+    }
+
+
+    @Override
+    public CommandResult execute() {
+        try {
+            final ReadOnlyPerson target = getTargetPerson();
+            addressBook.favouritePerson(target);
+            return new CommandResult(String.format(MESSAGE_FAVOURITE_PERSON_SUCCESS, target));
+        } catch (IndexOutOfBoundsException ie) {
+            return new CommandResult(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        } catch (PersonNotFoundException pnfe) {
```

Comment:
The favourite command is based on the index shown from a list of people who exists in the address book so would there be a PersonNotFoundException? Wouldn't the index out of bounds exception be enough if they enter an invalid index?

Reply:
Neh. He is catching for the case of having an outdated list. 
ie,
1.  You list 5 people
2.  You delete person 2
3.  You try to favourite person 2.

As the list is only updated per call to `list`, the list may be outdated and still contain non-existent people.
Good job catching for this edge case (How much is inherited, I will not comment)